spec/factories/course_assessment_categories.rb

```diff
@@ -2,7 +2,7 @@
   sequence(:course_assessment_category_weight) { |n| n }
   sequence(:course_assessment_category_title) { |n| "Category #{n}" }
   factory :course_assessment_category, class: Course::Assessment::Category do
-    course
+    course { create(:course) }
```

Comment:
Yep, this would be another way of doing it, sure.

Component callbacks is able to add normal AR callbacks, currently only after_create was hooked.